src/ECER.Resources.Documents/Applications/ApplicationRepository.cs

```diff
@@ -70,10 +70,26 @@ public async Task<string> SaveDraft(Application application)
     _ = UpdateApplicationTranscripts(ecerApplication, ecerTranscripts);
     _ = UpdateApplicationWorkExperienceReferences(ecerApplication, ecerWorkExperienceReferences);
     _ = UpdateCharacterReferences(ecerApplication, ecerCharacterReferences);
+
+    cancellationToken.ThrowIfCancellationRequested();
     context.SaveChanges();
     return ecerApplication.ecer_ApplicationId.Value.ToString();
   }
 
+  public async Task<string> Submit(string applicationId, CancellationToken cancellationToken)
+  {
+    await Task.CompletedTask;
+    var application = context.ecer_ApplicationSet.FirstOrDefault(d => d.ecer_ApplicationId == Guid.Parse(applicationId));
+    if (application == null) throw new InvalidOperationException($"Application '{applicationId}' not found");
+
+    application.StatusCode = ecer_Application_StatusCode.Submitted;
+    context.UpdateObject(application);
+
+    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
these are not needed - it may be important for long running operations, but this is not one